### PR TITLE
Add test cases of load and other functions

### PR DESCRIPTION
**Pull request by NicoYuan1986**
link: #23964
issue: #23989

### CODE REVIEW

File: tests/python_client/testcases/test_search.py [hunk 1]
```diff
@@ -3062,6 +3062,34 @@ def test_search_output_vector_field_and_scalar_field(self):
...
+                                                 default_string_field_name,
+                                                 default_search_field],
+                                  check_task=CheckTasks.check_search_results,
+                                  check_items={"nq": 1,
+                                               "limit": default_limit})[0]
+
+        # 3. check the result
+        assert default_int64_field_name, default_string_field_name in res[0][0].entity._row_data
+        assert default_search_field in res[0][0].entity._row_data
+        assert len(res[0][0].entity._row_data) == 3
```
Comment (binbinlv) [inline tests/python_client/testcases/test_search.py:3092]: Only length check is not enough either.

Try out_put = output_fields=[default_int64_field_name,
                                                 default_string_field_name,
                                                 default_search_field],

And out_put = output_fields=[default_string_field_name,
                                                 default_search_field],

And compare the results, you could know why.

And suggest to write an output_fields results check function to replace all the current check.

File: tests/python_client/testcases/test_collection.py [hunk 2]
```diff
@@ -2904,6 +2904,315 @@ def test_load_collection_without_creating_index(self):
                           check_items={"err_code": 1,
                                        "err_msg": "index not exist"})
 
+    @pytest.mark.tags(CaseLabel.L2)
+    def test_delete_load_collection_release_partition(self):
```
Comment (binbinlv) [inline tests/python_client/testcases/test_collection.py:2908]: I think search for all these scenario are needed too.

Comment (NicoYuan1986) [reply tests/python_client/testcases/test_collection.py:2908]: I have added search for all these scenarios.

File: tests/python_client/testcases/test_search.py (hunk 1 above)
Comment (NicoYuan1986) [reply tests/python_client/testcases/test_search.py:3092]: I will add the output_fields check function in another pr. Cause i found it hard to get the entity data. #24014

File: tests/python_client/testcases/test_search.py [hunk 3]
```diff
@@ -6660,3 +6687,540 @@ def test_range_search_with_consistency_session(self, nq, dim, auto_id, _async):
...
+    @pytest.mark.tags(CaseLabel.L2)
+    def test_load_partition_delete_drop_partition(self):
+        """
+        target: test load partition delete drop partition
+        method: 1. create a collection and 2 partitions
+                2. insert data
+                3. load one partition
+                4. delete half data in each partition
+                5. drop the non-loaded partition
+                6. search on the dropped partition -> len(res)==50
```
Comment (yanliang567) [inline tests/python_client/testcases/test_search.py:6902]: search on the non-dropped partition?

File: tests/python_client/testcases/test_search.py [hunk 4]
```diff
@@ -6660,3 +6687,540 @@ def test_range_search_with_consistency_session(self, nq, dim, auto_id, _async):
...
+        collection_w.get_compaction_state()
+        # load && release
+        collection_w.load()
+        partition_w1.release()
+        # search
+        collection_w.search(vectors[:1], default_search_field, default_search_params,
+                            default_limit, default_search_exp, partition_names=[partition_w1.name],
+                            check_task=CheckTasks.err_res,
+                            check_items={ct.err_code: 1, ct.err_msg: 'not loaded'})
+        res = collection_w.search(vectors[:1], default_search_field, default_search_params,
```
Comment (yanliang567) [inline tests/python_client/testcases/test_search.py:7051]: shall we add a search in the whole collection and check the results persist with results search in partition2

File: tests/python_client/testcases/test_search.py [hunk 5]
```diff
@@ -6660,3 +6687,540 @@ def test_range_search_with_consistency_session(self, nq, dim, auto_id, _async):
...
+        partition_w1.insert(df[100:200])
+        partition_w2.insert(df[200:300])
+        # compact
+        collection_w.compact()
+        collection_w.get_compaction_state()
+        # load && release
+        partition_w2.load()
+        collection_w.release()
+        # search
+        collection_w.search(vectors[:1], default_search_field, default_search_params,
```
Comment (yanliang567) [inline tests/python_client/testcases/test_search.py:7119]: what about if search parition_names=[partiton_w1, partition_w2] at this moment？

File: tests/python_client/testcases/test_search.py [hunk 6]
```diff
@@ -6660,3 +6687,540 @@ def test_range_search_with_consistency_session(self, nq, dim, auto_id, _async):
...
+        partition_w1.insert(df[100:200])
+        partition_w2.insert(df[200:300])
+        # load
+        collection_w.load()
+        # compact
+        collection_w.compact()
+        collection_w.get_compaction_state()
+        # release
+        partition_w2.release()
+        partition_w2.drop()
```
Comment (yanliang567) [inline tests/python_client/testcases/test_search.py:7151]: i think it worth a try of dropping the partition directly without release

File: tests/python_client/testcases/test_search.py (hunk 3 above)
Comment (NicoYuan1986) [reply tests/python_client/testcases/test_search.py:6902]: > search on the non-dropped partition?

Here should be search on the non-dropped partition. I'll correct this sentence.

File: tests/python_client/testcases/test_search.py (hunk 5 above)
Comment (NicoYuan1986) [reply tests/python_client/testcases/test_search.py:7119]: > what about if search parition_names=[partiton_w1, partition_w2] at this moment？

The results are same with collection.search(). This way of writing can be added in some cases.

File: tests/python_client/testcases/test_search.py (hunk 6 above)
Comment (NicoYuan1986) [reply tests/python_client/testcases/test_search.py:7151]: > i think it worth a try of dropping the partition directly without release

According to issue #23620, if take away this row ‘partition_w2.release()’, an error will be raised.

**Pull request merged**